Assets/Scripts/Noise/PerlinNoise.cs

```diff
@@ -0,0 +1,210 @@
+/// <summary>
+/// Using Perlin Noise function from adrian's soapbox "Understanding Perlin Noise" by Flafla2.
+/// https://flafla2.github.io/2014/08/09/perlinnoise.html
+/// From August 9th, 2014
+/// 
+/// github link: https://gist.github.com/Flafla2/f0260a861be0ebdeef76
+/// </summary>
+public class PerlinNoise {
+    /// <summary>
+    /// Doubled permutation to avoid overflow.
+    /// </summary>
+    private int[] p;
+
+    /// <summary>
+    /// Bounds the perlin noise to a grid space and repeats that grid square.
+    /// </summary>
+    private int repeat = 0;
+
+    /// <summary>
+    /// Makes a perlin noise generator with given settings.
+    /// </summary>
+    /// <param name="repeat">Bounds noise to a repeating square of size repeat by repeat. If this 
+    /// value is set to zero, there is no repeat.</param>
+    /// /// <param name="seed">Seed value for random permutation. if set to zero a arbitrary seed will be used.</param>
+    public PerlinNoise(int repeat, int seed) {
+        int[] permutation = new int[256];
+
+        for (int i = 0; i < 256; i++) {
+            permutation[i] = i;
+        }
+        
+        System.Random random = seed == 0 ? new System.Random() : new System.Random(seed);
+        int n = permutation.Length;
+        for (int i = n - 1; i > 1; i--) {
+            int rnd = random.Next(i+1);
+            int value = permutation[rnd];
+            permutation[rnd] = permutation[i];
+            permutation[i] = value;
+        }
+
+        p = new int[512];
+        for(int x=0;x<512;x++) {
+            p[x] = permutation[x%256];
+        }
+
+        this.repeat = repeat;
+    }
+
+    /// <summary>
+    /// Calculates the Perlin Noise value at a given x, y, z.
+    /// </summary>
+    /// <param name="x">X coordinate</param>
+    /// <param name="y">Y coordinate</param>
+    /// <param name="z">Z coordinate</param>
+    /// <returns>Perlin noise value at the given coordinate.</returns>
+    public float Perlin(float x, float y, float z) {
```

Comment:
Would it be better to take in a Vector here. Since vectors have x,y,z members?

Reply:
Yes, I changed it to use a Vector3. Theoretically we could upgrade this to use a n-dimensional tensor but that is for another day :)